gpii/node_modules/preferencesServer/src/preferencesPutHandler.js

```diff
@@ -51,12 +51,21 @@ https://github.com/GPII/universal/blob/master/LICENSE.txt
     });
 
     gpii.preferencesServer.put.request = function (request, preferencesServer, event) {
-        preferencesServer.getRawPreferences(event, request.req.params.userToken);
+        preferencesServer.getRawPreferencesAllowError(event, request.req.params.userToken);
     };
 
-    gpii.preferencesServer.put.onGetRawPreferences = function (preferencesServer, ontologyHandler, request, rawPrefs, event) {
+    gpii.preferencesServer.put.onGetRawPreferences = function (preferencesServer, ontologyHandler, request, requestProxy, rawPrefs, event) {
```

Comment:
Remove requestProxy here. Also, can you refactor the production of rawPrefs so that it is a proper promise, rather than being a "prefs or error" object as it currently is. Even if we have a fault in Kettle, we should try to correct it as early in the handling chain as possible so we don't have this redundant code everywhere.